.ci/release/Jenkinsfile

```diff
@@ -0,0 +1,235 @@
+                // Construct the URL with anchor for the release notes
+                // Ex: https://www.elastic.co/guide/en/apm/agent/java/current/release-notes-1.x.html#release-notes-1.13.0
+                def baseUrl = "https://www.elastic.co/guide/en/apm/agent/java/current/release-notes-"
+                def dotX = sh(script: 'git tag|tail -1|sed s/v//|cut -f1 -d "."|awk \'{print $1".x"}\'', returnStdout: true)
+                def bareTag = sh(script: 'git tag|tail -1|sed s/v//', returnStdout: true)
+                def finalUrl = sh(script: "echo $baseUrl$dotX\.html#release-notes-$bareTag")
+                def ret = githubReleaseCreate(draft: true, body: "[Release Notes for $bareTag]($finalUrl)")
+                env.RELEASE_ID = ret['id']
+              }
+            }
+
+          }
+        }
+        // 10. Update cloudfoundry/index.yml
+        // Needs to append a line such as the following:
+        // 1.13.0: https://repo1.maven.org/maven2/co/elastic/apm/elastic-apm-agent/1.13.0/elastic-apm-agent-1.13.0.jar
+        stage('Update Cloudfoundry') {
+          steps {
+            dir("${basedir}") {
+              script {
+                sh("git checkout master")
+                def bareTag = sh(script: 'git tag|tail -1|sed s/v//', returnStdout: true)
+                def baseUrl = "https://repo1.maven.org/maven2/co/elastic/apm/elastic-apm-agent"
+                def lineToAppend = sh(script: "echo '$baseUrl: $baseUrl/$bareTag/elastic-apm-agent-$bareTag\.jar' >> cloudfoundry/index.yml ")
+                gitPush()
+              }
+            }
+          }
+        }
+        // 11. Wait for released package to be available in maven central
+        stage('Wait for artifact to be available in Maven Central') {
+          steps {
+            dir("${BASE_DIR}"){
+              script {
+                // We are looking for something like this:
+                // https://oss.sonatype.org/service/local/repositories/releases/content/co/elastic/apm/apm-agent-java/1.1.12/apm-agent-java-1.1.12.pom
+                def baseUrl = "https://oss.sonatype.org/service/local/repositories/releases/content/co/elastic/apm/apm-agent-java"
+                def bareTag = sh(script: 'git tag|tail -1|sed s/v//', returnStdout: true)
+                def fullUrl = sh(script: "$baseUrl/$bareTag/apm-agent-java-$bareTag\.pom")
+                waitUntil(initialRecurrencePeriod: 15000) {
+                  script {
+                    def ret = sh(script: "curl -fs $fullUrl >/dev/null 2>&1", returnStatus: true)
+                    echo "Waiting for the artifacts to be published on Sonatype"
+                    return ret == 0
+                  }
+                }
+              }
+            }
+
+          }
+        }
+        // 12. Publish release on Github. This will notify users watching repository.
+        stage('Publish release on GitHub') {
+          steps {
+            githubReleasePublish(id: ${env.RELEASE_ID})
```

Comment:
The GH release contains a link to a release-notes URL calculated in the `Create GitHub release draft` stage. This link will be available after the relevant branch gets built, after pushing the new tag in the `Branch creation` phase.
It takes time, so this step needs to block until the release-notes URL is available, otherwise we will publish a Release with a broken link.